sql/core/src/test/scala/org/apache/spark/sql/DatasetCacheSuite.scala

```diff
@@ -132,4 +132,19 @@ class DatasetCacheSuite extends QueryTest with SharedSQLContext with TimeLimits
     df.unpersist()
     assert(df.storageLevel == StorageLevel.NONE)
   }
+
+  test("SPARK-24613 Cache with UDF could not be matched with subsequent dependent caches") {
+    val expensiveUDF = udf({x: Int => Thread.sleep(10000); x})
```

Comment:
can we use accumulators and make sure this UDF only run 10 times? sleeping 10 seconds is not good in a unit test

Reply:
Accumulators probably wouldn't work. I'll do verify plan though.